src/libraries/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/ComInterop/VariantArray.cs

```diff
@@ -49,11 +50,23 @@ internal static class VariantArray
         // (guarenteed less than 28, in practice 0-2)
         private static readonly List<Type> s_generatedTypes = new List<Type>(0);
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray1))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray2))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray4))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray8))]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "Types are either dynamically created or have dynamic dependency.")]
         internal static MemberExpression GetStructField(ParameterExpression variantArray, int field)
         {
             return Expression.Field(variantArray, "Element" + field);
         }
 
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray1))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray2))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray4))]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(VariantArray8))]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2055:UnrecognizedReflectionPattern",
+            Justification = "Types are either dynamically created or have dynamic dependency.")]
```

Comment:
I don't believe the `DynamicDependency` attributes on this method are resolving anything, and can be removed. The Justification here is 
```suggestion
            Justification = "MakeGenericType is called on a dynamically created type that doesn't contain trimming annotations.")]
```